apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts

```diff
@@ -199,374 +170,533 @@ describe('FilesStorageUC', () => {
+					allowedType,
+					params.parentId,
+					PermissionContexts.create
 				);
 			});
 
-			it('should throw Error', async () => {
-				authorizationService.checkPermissionByReferences.mockRejectedValue(new ForbiddenException());
-				await expect(filesStorageUC.upload(userId, fileUploadParams, request)).rejects.toThrow();
-			});
-		});
+			it('should call uploadFile with correct params', async () => {
+				const { params, userId, request, buffer, fileInfo } = setup();
+
+				await filesStorageUC.upload(userId, params, request);
+
+				const fileDescription: IFile = IFileBuilder.buildFromRequest(fileInfo, request, buffer);
 
-		describe('Error Handling()', () => {
-			beforeEach(() => {
-				storageClient.create.mockRejectedValue(new Error());
+				expect(filesStorageService.uploadFile).toHaveBeenCalledWith(userId, params, fileDescription);
 			});
 
-			it('should throw Error', async () => {
-				await expect(filesStorageUC.upload(userId, fileUploadParams, request)).rejects.toThrow();
-			});
-
-			it('should call fileRecordRepo.removeAndFlush', async () => {
-				await expect(filesStorageUC.upload(userId, fileUploadParams, request)).rejects.toThrow();
-
-				expect(fileRecordRepo.delete).toBeCalledWith(
-					expect.objectContaining({
-						id: entityId,
-						name: 'text (1).txt',
-						size: 1234,
-						parentType: FileRecordParentType.User,
-						mimeType: 'text/plain',
-						createdAt: expect.any(Date) as Date,
-						updatedAt: expect.any(Date) as Date,
-					})
-				);
+			it('should call uploadFile with correct params', async () => {
+				const { params, userId, request, fileRecord } = setup();
+
+				const result = await filesStorageUC.upload(userId, params, request);
+
+				expect(result).toEqual(fileRecord);
 			});
 		});
-	});
 
-	describe('download()', () => {
-		beforeEach(() => {
-			fileRecordRepo.findOneById.mockResolvedValue(fileRecord);
-			storageClient.get.mockResolvedValue(response);
-		});
+		describe('WHEN user is authorized and busboy emits error', () => {
+			const setup = () => {
+				const { params, userId } = getFileRecordsWithParams();
+				const request = getRequest();
+				const error = new Error('test');
 
-		describe('calls to fileRecordRepo.findOneById()', () => {
-			it('should call once', async () => {
-				await filesStorageUC.download(userId, fileDownloadParams);
-				expect(fileRecordRepo.findOneById).toHaveBeenCalledTimes(1);
-			});
+				const mockBusboyEvent = (requestStream: DeepMocked<Busboy>) => {
+					requestStream.emit('error', error);
 
-			it('should call with fileRecordId', async () => {
-				await filesStorageUC.download(userId, fileDownloadParams);
-				expect(fileRecordRepo.findOneById).toBeCalledWith(fileDownloadParams.fileRecordId);
+					return requestStream;
+				};
+
+				const size = request.headers['content-length'];
+				request.get.mockReturnValue(size);
+				request.pipe.mockImplementation(mockBusboyEvent as never);
```

Comment:
as never

Reply:
With direct use of callback in mockImplementation never is not needed anymore